src/datasets/transformations.jl

```diff
@@ -157,6 +161,36 @@ getobs(jdata, 15) == 15
 """
 joinobs(datas...) = JoinedData(datas)
 
+"""
+    tokenize(type, input)
+    type = :words or :chars
+
+Tokenizes an input string or stream into pieces depending on selected type. 
+"""
+
+struct Tokenizer
+    data::AbstractArray{AbstractString}
+end
+
+function tokenize(type,input)
+    ts = TokenBuffer(input)
+    if type === :chars
+        while !isdone(ts)
+            character(ts)
+            flush!(ts)
+        end
+    elseif type === :words
+        tokenize(input)
+    end
+    return Tokenizer(ts.tokens)
+end
+
+# LearnBase functions {nobs, getobs} for Tokens
+
+LearnBase.nobs(toks::Tokenizer) = length(toks.data)
+function LearnBase.getobs(toks::Tokenizer, idx)
+    return toks.data[idx]
+end
```

Comment:
```suggestion
LearnBase.getobs(toks::Tokenizer, idx) = toks.data[idx]
```
Just to make things cleaner